### PR TITLE
Java 21 migration changes

### PR DESCRIPTION
**Pull request by dgurjar-ch**
Update spring boot 3 dependency
Update java 21 jar
Configured HealthChecks
Updated httpcomponent jar
Updated maven plugin
Removed duplicated code

### CODE REVIEW

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java [hunk 1]
```diff
@@ -1,9 +1,9 @@
 package uk.gov.companieshouse.document.generator.accounts.handler.accounts;
 
+import org.json.JSONObject;
```
Comment (vshastri-ch) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java:3]: move it to the group of org.* imports.

File: document-generator-company-report/pom.xml [hunk 2]
```diff
@@ -10,6 +10,16 @@
     <modelVersion>4.0.0</modelVersion>
 
     <artifactId>document-generator-company-report</artifactId>
+    <properties>
+        <!-- Encoding -->
```
Comment (vshastri-ch) [inline document-generator-company-report/pom.xml:14]: these 2 comments are not needed as its self explanatory

File: pom.xml [hunk 3]
```diff
@@ -169,28 +193,25 @@
...
                             <artifactId>jib-maven-plugin</artifactId>
-                            <version>3.1.1</version>
+                            <version>${maven-jar-plugin.version}</version>
                             <configuration>
+                                <from>
+                                    <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/ci-corretto-build-21:latest</image>
+                                </from>
                                 <to>
-                                    <image>169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/document-generator</image>
+                                    <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/local/document-generator:latest</image>
```
Comment (vshastri-ch) [inline pom.xml:211]: why is it changed from 169942020521 to 416670754337?

File: document-generator-api/pom.xml [hunk 4]
```diff
@@ -21,11 +21,11 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         
         <!-- JDK Version -->
-        <jdk.version>1.8</jdk.version>
+        <jdk.version>21</jdk.version>
 
-        <apache-http-components.version>4.5.13</apache-http-components.version>
+        <apache-http-components.version>5.4-beta1</apache-http-components.version>
```
Comment (vshastri-ch) [inline document-generator-api/pom.xml:26]: Please double check if we are ok to have beta versions. I'd say go for the latest stable version.

**Review by hwatkins-ch — APPROVED** [review]
LGTM once comments sorted

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java (hunk 1 above)
Comment (dgurjar-ch) [reply document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java:3]: Done

File: document-generator-api/pom.xml (hunk 4 above)
Comment (dgurjar-ch) [reply document-generator-api/pom.xml:26]: Changed to 5.4

File: pom.xml (hunk 3 above)
Comment (dgurjar-ch) [reply pom.xml:211]: Actually in docker-composer had this so just replaced , Now changed to 169942020521

File: document-generator-company-report/pom.xml (hunk 2 above)
Comment (dgurjar-ch) [reply document-generator-company-report/pom.xml:14]: these are old comments , now removed.

**Review by vshastri-ch — APPROVED** [review]

**Pull request merged**